var/spack/repos/builtin/packages/emacs/package.py

```diff
@@ -77,3 +79,32 @@ def configure_args(self):
             args.append('--without-gnutls')
 
         return args
+
+    def _test_check_versions(self):
+        """Perform version checks on installed package binaries."""
+        spec_vers = str(self.spec.version)
+
+        checks = {
+            'ctags':        ([spec_vers], None),
+            'ebrowse':      ([spec_vers], None),
+            'emacs':        ([spec_vers], None),
+            'emacsclient':  ([spec_vers], None),
+            'etags':        ([spec_vers], None),
+        }
+
+        for exe in checks:
+            expected, status = checks[exe]
+            reason = 'test version of {0} is {1}'.format(exe, expected[0])
+            self.run_test(exe, ['--version'], expected, status, installed=True,
+                          purpose=reason, skip_missing=True)
+
+    def test(self):
+        """Perform smoke tests on the installed package."""
+        tty.warn('Expected results currently based on simple cmake builds')
+
+        if not self.spec.satisfies('@24.5:26.3'):
+            tty.warn('Expected results have not been confirmed for {0} {1}'
+                     .format(self.name, self.spec.version))
```

Comment:
Can you turn these warnings into comments (or debug statements if you feel strongly about showing them to the user) so we're less verbose?

Reply:
Sure, once I figure out how to deal with the `features/spack-test` divergence issue.

Reply:
Done.